bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -86,28 +89,44 @@
      * After the token is "expired," the interceptor blocks gRPC calls. The Expired state indicates
      * that the interceptor needs to do a synchronous refresh.
      */
-    public static final int TOKEN_EXPIRES_MS = 45 * 1000;
+    public static final int TOKEN_EXPIRES_MS =5 * 1000;
 
     final IOException exception;
     final String header;
-    final long staleTimeMs;
-    final long expiresTimeMs;
+    
+    /**
+     * Defines the amount of time in ms when the header is considered "stale" and should be
+     * refreshed in the near future.  A null value means that the header does not become stale.
+     */
+    final @Nullable Long staleTimeMs;
+
+    /**
+     * Defines the amount of time in ms when the header is considered "expired" and must be
+     * refreshed.  A null value means that the header does not expire.
+     */
+    final @Nullable Long expiresTimeMs;
 
     public HeaderCacheElement(AccessToken token) {
       this.exception = null;
       this.header = "Bearer " + token.getTokenValue();
-      long tokenExpiresTime = token.getExpirationTime().getTime();
-      this.staleTimeMs = tokenExpiresTime - TOKEN_STALENESS_MS;
-      // Block until refresh at this point.
-      this.expiresTimeMs = tokenExpiresTime - TOKEN_EXPIRES_MS;
-      Preconditions.checkState(staleTimeMs < expiresTimeMs);
+      Date expirationTime = token.getExpirationTime();
+      if (expirationTime != null) {
+        long tokenExpiresTime = expirationTime.getTime();
+        this.staleTimeMs = tokenExpiresTime - TOKEN_STALENESS_MS;
+        // Block until refresh at this point.
+        this.expiresTimeMs = tokenExpiresTime - TOKEN_EXPIRES_MS;
+        Preconditions.checkState(staleTimeMs < expiresTimeMs);
+      } else {
+        this.staleTimeMs = null;
```

Comment:
add a test case to test when these are set to null, header doesn't become stale or expired?